src/app/services/wiseLinkService.ts

```diff
@@ -7,42 +7,63 @@ import { UtilService } from '../../assets/wise5/services/utilService';
 @Injectable()
 export class WiseLinkService {
   constructor(
-    private NodeService: NodeService,
+    private nodeService: NodeService,
     private sanitizer: DomSanitizer,
-    private StudentDataService: StudentDataService,
-    private UtilService: UtilService
+    private studentDataService: StudentDataService,
+    private utilService: UtilService
   ) {}
 
-  createWiseLinkClickedHandler(currentNodeId: string): any {
+  wiseLinkClickedEventName: string = 'wiselinkclicked';
+  wiseLinkClickedHandler: any;
+  wiseLinkCommunicatorId: string = 'wise-link-communicator';
+
+  addWiseLinkClickedListener(): void {
+    this.wiseLinkClickedHandler = this.createWiseLinkClickedHandler();
+    this.getWiseLinkCommunicator().addEventListener(
+      this.wiseLinkClickedEventName,
+      this.wiseLinkClickedHandler
+    );
+  }
+
+  private createWiseLinkClickedHandler(): any {
+    const thisStudentDataService = this.studentDataService;
     return (event: CustomEvent) => {
+      const currentNodeId = thisStudentDataService.getCurrentNodeId();
       this.followLink(currentNodeId, event.detail.nodeId, event.detail.componentId);
     };
   }
 
-  addWiseLinkClickedListener(wiseLinkCommunicator: any, wiseLinkClickedHandler: any): void {
-    wiseLinkCommunicator.addEventListener('wiselinkclicked', wiseLinkClickedHandler);
+  removeWiseLinkClickedListener(): void {
+    this.getWiseLinkCommunicator().removeEventListener(
+      this.wiseLinkClickedEventName,
+      this.wiseLinkClickedHandler
+    );
+  }
+
+  private getWiseLinkCommunicator(): any {
+    return document.getElementById(this.wiseLinkCommunicatorId);
   }
 
-  followLink(currentNodeId: string, nodeId: string, componentId: string): void {
+  private followLink(currentNodeId: string, nodeId: string, componentId: string): void {
     if (nodeId === currentNodeId) {
-      this.NodeService.scrollToComponentAndHighlight(componentId);
+      this.nodeService.scrollToComponentAndHighlight(componentId);
     } else {
       this.goToNode(nodeId, componentId);
     }
   }
```

Comment:
Change this func to lookup currentNodeId and remove need to pass it in as param?
```
  private followLink(nodeId: string, componentId: string): void {
    if (nodeId === this.studentDataService.getCurrentNodeId()) {
      this.nodeService.scrollToComponentAndHighlight(componentId);
    } else {
      this.goToNode(nodeId, componentId);
    }
  }
```